gdk/common/exceptions/InitError.py

```diff
@@ -0,0 +1,37 @@
+class InitException(Exception):
+    def __init__(self, reason="", solution="", exception=None, help_text=""):
+        err_details = ""
+        if exception:
+            err_details = f"Error details: {exception}\n"
+        self.message = f"{reason}\n{err_details}{solution}{help_text}"
+        super().__init__(self.message)
+
+
+class InvalidArgumentsException(InitException):
```

Comment:
This could be reused for other commands as well

Reply:
This has `init` specific error reason and solution. Just renamed it for now. This is more like incomplete args passed to the command. 

This can be refactored but invalid args are actually caught by argparser. 

`gdk component init` is valid for argparser by not for gdk as we expect either template or repo args.

Reply:
error reason could be taken as argument to exception, so we dont need to create new exception if message needs to be customized. If exception needs to be handled same way, it should be reused.